### DISCUSSION TITLE
aws ce get-cost-and-usage-with-resources --granularity MONTHLY returns 'for hourly granularity is 14 days' massage.

### DISCUSSION THREAD

**Discussion by suito-venus** [aws/aws-cli#7382] [Q&A] [answered]
Hi,

I exec 'aws ce get-cost-and-usage-with-resources' command with --granularity MONTHLY option.
But command returned `start date is too old for hourly, the max supported days for hourly granularity is 14 days` error message.

Why and what is wrong?

```bash
$ aws ce get-cost-and-usage-with-resources  \
> --granularity MONTHLY \
> --time-period Start=2022-06-01,End=2022-09-01 \
> --filter '{"Dimensions": { "Key": "INSTANCE_TYPE", "Values": [ "UsageQuantity", "BlendedCosts", "RESOURCE_ID" ] }}' \
> --metrics "BlendedCost" "UsageQuantity" \
> --group-by Type=DIMENSION,Key=INSTANCE_TYPE Type=DIMENSION,Key=RESOURCE_ID \
> --profile mfa

An error occurred (ValidationException) when calling the GetCostAndUsageWithResources operation: start date is too old for hourly, the max supported days for hourly granularity is 14 days
$ aws --version
aws-cli/2.8.7 Python/3.9.11 Linux/5.10.16.3-microsoft-standard-WSL2 exe/x86_64.ubuntu.20 prompt/off
$
```
Thanks.

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @suito-venus, thanks for reaching out. 

I researched the error you're getting and found out in the [API documentation](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetCostAndUsageWithResources.html#API_GetCostAndUsageWithResources_RequestSyntax) that the start date has to be within past 14 days. 

>TimePeriod
>Sets the start and end dates for retrieving Amazon Web Services costs. **The range must be within the last 14 days (the start date cannot be earlier than 14 days ago).** The start date is inclusive, but the end date is exclusive. 

The required date range seems to ignore `granularity` as even if you're requesting `MONTHLY`, it wouldn't work unless you enter the date within 14 days ago. For example, the command below would work since the range is within past 14 days:

```
aws ce get-cost-and-usage-with-resources  \
--time-period '{"Start":"2022-10-25","End":"2022-10-31"}' \
--granularity 'MONTHLY' --filter '{"Dimensions": { "Key": "INSTANCE_TYPE", "Values": [ "UsageQuantity", "BlendedCosts", "RESOURCE_ID" ] }}' \
--metrics "BlendedCost" "UsageQuantity" \
--group-by Type=DIMENSION,Key=INSTANCE_TYPE Type=DIMENSION,Key=RESOURCE_ID \
```

I understand that it can be confusing and that the behavior is inconsistent with the documentation which is why I reached out to Cost Explorer and created this [Github issue](https://github.com/aws/aws-sdk/issues/381) in our cross-sdk repository.  Please refer to that issue for tracking the request going forward. Feel free to leave any feedback there and I can pass it along to the Cost Explorer team.

**Reply by suito-venus** [reply]
Hi, @aBurmeseDev .
Thanks for your reply, and pass it to the Cost Explorer team! 

I will watch Github issue!
